### ISSUE TITLE
Warn for extra semicolon

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust#56338]
The closing semicolon of the match {} is not warned against by Rusts nor Clippy:

```
enum Foo { A, B }

fn bar(f: Foo) {
    match f {
        Foo::A => {},
        Foo::B => {},
    };
}

fn main() {
    bar(Foo::A);
    bar(Foo::B);
}
```

**Comment by killercup** [comment]
Not sure this is lint-worthy as it doesn't change the semantics of the code. Is it removed by rustfmt?

**Comment by leonardo-m** [comment]
> Is it removed by rustfmt?

Currently rustfmt doesn't remove it.

> Not sure this is lint-worthy as it doesn't change the semantics of the code.

From my experience when the code isn't clean, it could cause troubles for someone later. I prefer Rust code to be tidy, it causes less troubles.

**Comment by brunoczim** [comment]
I don't think this should be a rustc's lint. It is an insignificant syntactic detail. There are much worse things like lack of indentation. I don't think we should warn those things either. Maybe it could be a clippy's lint (of pedant category). 

Currently, rustfmt cannot remove this extra semicolon because it requires semantical analysis and it only does syntactic analysis. We need to know if the type of the expression is `()`.

**Comment by leonardo-m** [comment]
If that's the consensus I'll move this issue to Clippy issues.

**Comment by estebank** [comment]
@leonardo-m closing and linking from [similar clippy issue](https://github.com/rust-lang/rust-clippy/issues/3305).
